extensions/scarb-doc/src/types/groups.rs

```diff
@@ -20,6 +20,7 @@ pub struct Group<'db> {
     pub impls: Vec<Impl<'db>>,
     pub extern_types: Vec<ExternType<'db>>,
     pub extern_functions: Vec<ExternFunction<'db>>,
+    pub macro_declarations: Vec<MacroDeclaration<'db>>,
```

Comment:
macro declaration cannot have a group attribute but the field is added for the sake of macro rule [`define_insert_function`](https://github.com/software-mansion/scarb/blob/036db00740c43fe376a157f4c99a9ab4936ded52/extensions/scarb-doc/src/types/module_type.rs#L171C14-L171C36) - I'm not sure about this so let me know if you think it's a bad idea. 
Another thing would be that it keeps the option open for adding the group attribute support for macro definitions (not sure what are the perspectives here).

Reply:
TBH I don't have strong opinion, but seems okay to me